gems/pending/openstack/openstack_handle/network_delegate.rb

```diff
@@ -15,7 +15,15 @@ def security_groups_for_accessible_tenants
     end
 
     def quotas_for_current_tenant
-      @tenant_id ||= current_tenant['id']
+      if current_tenant.kind_of?(Hash)
+        @tenant_id ||= current_tenant['id']
+      else
+        # Seems like keystone v3 has string in current_tenant
+        @tenant_id = @os_handle.accessible_tenants.detect { |x| x.name == current_tenant }.try(:id)
+      end
```

Comment:
Maybe this code should be pulled up into the Handle directly?  Then here it would just be:

```ruby
   @tenant_id ||= @os_handle.current_tenant
```

Looks like this code is gonna be in about all of the delegates, otherwise.